backend/dal/Services/Concrete/ProjectService.cs

```diff
@@ -780,35 +801,28 @@ public async System.Threading.Tasks.Task<Project> SetStatusAsync(Project project
             project.CopyRowVersionTo(originalProject);
             this.Context.CommitTransaction();
 
+            await SendNotificationsAsync(originalProject, fromStatusId, responses, noteChanged);
+
             return Get(originalProject.Id);
         }
 
         #region Notifications
         /// <summary>
-        /// Cancel any pending notifications for the specified project 'id'.
-        /// </summary>
-        /// <param name="id"></param>
-        /// <returns></returns>
-        public async System.Threading.Tasks.Task<Paged<NotificationQueue>> CancelNotificationsAsync(int id)
-        {
-            var page = GetNotificationsInQueue(new ProjectNotificationFilter() { Quantity = 500, ProjectId = id, Status = new[] { NotificationStatus.Accepted, NotificationStatus.Pending } });
-            await this.Self.NotificationQueue.CancelNotificationsAsync(page);
-            return page;
-        }
-
-        /// <summary>
-        /// Cancel any pending notifications for the specified project 'id' and agency 'agencyId'.
+        /// Cancel all the pending notifications for the specified 'projectId' and 'agencyId'.
+        /// This function does not update the data source.
         /// </summary>
-        /// <param name="id"></param>
+        /// <param name="projectId"></param>
         /// <param name="agencyId"></param>
         /// <returns></returns>
-        public async System.Threading.Tasks.Task<Paged<NotificationQueue>> CancelNotificationsAsync(int id, int agencyId)
+        public async System.Threading.Tasks.Task<IEnumerable<NotificationQueue>> CancelNotificationsAsync(int projectId, int? agencyId = null)
```

Comment:
I don't like having the cancel notification logic in the DAL, but couldn't think of a better way at the moment.